File: models/Gpt-3.5-turbo/cpp/code/problems/sorting_and_searching/Tasks_and_Deadlines.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<pair<int, int>> tasks(n);
    for (int i = 0; i < n; ++i) {
        cin >> tasks[i].first >> tasks[i].second;
    }

    sort(tasks.begin(), tasks.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
        return a.second < b.second;
    });

    int time = 0;
    int reward = 0;
    for (int i = 0; i < n; ++i) {
        time += tasks[i].first;
        if (time > tasks[i].second) {
            reward -= (time - tasks[i].second);
        }
    }

    cout << reward << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/sorting_and_searching/Tasks_and_Deadlines.py

# Read the input
n = int(input())
tasks = []
for _ in range(n):
    a, d = map(int, input().split())
    tasks.append((a, d))

# Sort the tasks in decreasing order of deadlines
tasks.sort(key=lambda x: x[1], reverse=True)

# Initialize the variables
reward = 0
time = 0

# Process the tasks one by one
for task in tasks:
    a, d = task
    # Check if we can complete the task before its deadline
    if time + a <= d:
        reward += d - (time + a)
        time += a

# Print the maximum reward
print(reward)
